tutorials/calibration_flow/gptq_like.py

```diff
@@ -0,0 +1,273 @@
+            for inp in grouped_args:
+                # inp = tensors_to_cuda(inp)
+                cur_args, cur_kwargs = tree_unflatten(inp, spec)
+                out = func(*cur_args, **cur_kwargs)
+                # outputs.append(out.cpu() if isinstance(out, torch.Tensor) else out)
+                outputs.append(out)
+            grouped_outputs = [tree_flatten(x)[0] for x in outputs]
+            out_spec = tree_flatten(outputs[0])[1]
+            # convert [[A,b1,c1], [A,b2,c2] [A,b3,c3]] => [A, MultiTensor(b1,b2,b3), MultiTensor(c1,c2,c3)]
+            flat_outputs, non_tensors_equal = grouped_to_flat(grouped_outputs)
+            assert non_tensors_equal, (
+                f"ERR: found a function in model: {func} which "
+                +"caused an error in MultiInput, the function dispatch only works for functions"
+                +" with Tensor outputs or that have the same non-Tensor output value for all across all inputs"
+            )
+            return tree_unflatten(flat_outputs, out_spec)
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs={}, skip_gptq=False):
+        pass
+
+    def __tensor_flatten__(self):
+        return ["values"], None
+
+    @classmethod
+    def __tensor_unflatten__(
+        cls, tensor_data_dict, tensor_attributes, outer_size, outer_stride
+    ):
+        return cls(tensor_data_dict["values"])
+
+class TwoLinear(torch.nn.Module):
+    def __init__(self, in_features=64, out_features=128):
+        super().__init__()
+        self.linear1 = torch.nn.Linear(in_features, out_features)
+        self.linear2 = torch.nn.Linear(in_features, out_features)
+
+    def forward(self, x, y):
+        x = self.linear1(x)
+        y = self.linear2(y)
+        return x + y
+
+class M(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.two_linear1 = TwoLinear()
+        self.two_linear2 = TwoLinear(128, 256)
+
+    def forward(self, x, y):
+        x1 = self.two_linear1(x, y)
+        x2 = self.two_linear2(x1, x1)
+        return x2
+
+def _is_two_linear(mod, fqn):
+    return isinstance(mod, TwoLinear)
+
+# Adapted from https://github.com/pytorch/ao/pull/581
+def prepare_model_for_optimization_(model):
+    def forward_hook(
+        module,
+        args: Tuple[MultiTensor],
+        kwargs: Dict[str, MultiTensor],
+        output: Tuple[MultiTensor],
+    ):
+        # remove the hook to avoid recursive calls
+        module._forward_hook_handle.remove()
+        # in this case args will be a tuple of 2 MultiTensors since
+        # the TwoLinear module takes 2 inputs, and each input will collect a list of normal Tensors
+
+        # we can use these two MultiTensors to calculate the quantization parameters for each input Tensor
+        act_obs1 = AffineQuantizedMinMaxObserver(MappingType.ASYMMETRIC, torch.uint8, granularity_type=PerTensor(), eps=torch.finfo(torch.float32).eps, scale_dtype=torch.float32, zero_point_dtype=torch.int32)
```

Comment:
I'm still not sure I follow why the hello world example for static quantization should use 2 modules? Also the TwoLinear model seems kinda contrived and more of a test instead of a tutorial?

Reply:
yeah it's a bit more complicated example, since I want to show the output of the module is modified in `        output_with_fake_quantized_inputs = module(mt1, mt2)`, I can change to one input and one module as well if you feel it's more important to keep it simple